asset/mock.go

```diff
@@ -61,6 +65,160 @@ func RandGroupKeyWithSigner(t testing.TB, genesis Genesis) (*GroupKey, []byte) {
 
 	return groupKey, privateKey.Serialize()
 }
+
+// Copied from tapscript/tx.go.
+func virtualGenesisTxWithInput(virtualTx *wire.MsgTx, input *Asset,
+	idx uint32, witness wire.TxWitness) *wire.MsgTx {
+
+	txCopy := virtualTx.Copy()
+	txCopy.LockTime = uint32(input.LockTime)
+	// Using the standard zeroIndex for virtual prev outs.
+	txCopy.TxIn[0].PreviousOutPoint.Index = idx
+	txCopy.TxIn[0].Sequence = uint32(input.RelativeLockTime)
+	txCopy.TxIn[0].Witness = witness
+	return txCopy
+}
+
+// Copied from tapscript/tx.go.
+func inputGenesisAssetPrevOut(prevAsset Asset) (*wire.TxOut, error) {
+	switch prevAsset.ScriptVersion {
+	case ScriptV0:
+		// If the input asset is a genesis asset that is part of an
+		// asset group, we need to validate the group witness against
+		// the tweaked group key and not the genesis asset script key.
+		validationKey := &prevAsset.GroupKey.GroupPubKey
+		pkScript, err := test.ComputeTaprootScriptErr(
+			schnorr.SerializePubKey(validationKey),
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		return &wire.TxOut{
+			Value:    int64(prevAsset.Amount),
+			PkScript: pkScript,
+		}, nil
+	default:
+		return nil, ErrUnknownVersion
+	}
+}
+
+// Forked from tapscript/tx/virtualTxIn to remove checks for non-nil PrevIDs on
+// inputs and full consumption of inputs.
+func VirtualGenesisTxIn(newAsset *Asset) (*wire.TxIn, mssmt.Tree, error) {
```

Comment:
I think we can move this out of `mock.go`, so residing in like `asset/group_genesis.go` or something like that. After we get things all working we maybe just want to move it purely within `txscript`, then we keep the `asset` package concerned with mainly serialization and initial sanity checks.

Reply:
Moved in latest version out to `asset/asset` near the other group logic - I agree that we probably want to rearrange these implementations overall.